lib/gather/collector.rb

```diff
@@ -0,0 +1,136 @@
+#==============================================================================
+# Copyright (C) 2023-present Alces Flight Ltd.
+#
+# This file is part of Flight Gather.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# Flight Gather is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with Flight Gather. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Gather, please visit:
+# https://github.com/openflighthpc/flight-gather
+#==============================================================================
+
+require 'socket'
+require 'yaml'
+
+module Gather
+  module Collector
```

Comment:
This is likely better suited for the `class` namespace. Judging by how it's being called in `collect`...

```ruby
if @options.type.nil? || @options.type.downcase == 'physical'
  puts 'Gathering physical data...'
  data = data.deep_merge(Collector.physical_data)
end

if @options.type.nil? || @options.type.downcase == 'logical'
  puts 'Gathering logical data...'
  data = data.deep_merge(Collector.logical_data)
end
```

...you're already treating it like a class. A module is designed for abstracting shared methods across classes, and would be used like this:

```ruby
module Collector
  def logical
    "really important data relating to #{self.class}"
  end
end

class System
  include Collector

  def initialize
    @logical = logical
    super
  end
end
```